src/c_kzg_4844.c

```diff
@@ -719,13 +716,14 @@ static void compute_challenge(
     uint8_t *offset = bytes;
 
     /* Copy domain separator */
-    memcpy(offset, FIAT_SHAMIR_PROTOCOL_DOMAIN, 16);
-    offset += 16;
+    memcpy(offset, FIAT_SHAMIR_PROTOCOL_DOMAIN, BYTES_PER_DOMAIN);
+    offset += BYTES_PER_DOMAIN;
+
+    /* Copy polynomial degree (16-bytes, little-endian) */
     bytes_from_uint64(offset, FIELD_ELEMENTS_PER_BLOB);
-    offset += 8;
-    /* Set all other bytes of this 16-byte (little-endian) field to zero */
+    offset += sizeof(uint64_t);
```

Comment:
Sounds good! Although I think this change is not that useful if we don't include `sizeof(uint64_t)` in the definition of `CHALLENGE_INPUT_SIZE` as well.

Reply:
Included `sizeof(uint64_t)` in the definition, part of `DOMAIN_SEPARATOR_LENGTH`.